### ISSUE TITLE
TypeError: CaptureTwistedLogs puts strings into details

### ISSUE THREAD

**Issue by allenap** [testing-cabal/testtools#230]
Using testtools 2.1.0 with Python 3.5.1 on Ubuntu 16.04 I observe:

``` python
>>> from testtools.twistedsupport import CaptureTwistedLogs
>>> ctl = CaptureTwistedLogs()
>>> ctl.setUp()
>>> ctl.getDetails()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/gavin/.buildout/eggs/testtools-2.1.0-py3.5.egg/testtools/content.py", line 123, in __repr__
    self.content_type, _join_b(self.iter_bytes()))
TypeError: sequence item 0: expected a bytes-like object, str found
```

The iterable of bytes given to `Content` contains a Unicode string:

``` python
>>> ctl.getDetails()["twisted-log"].iter_bytes()
['']
```

I think this is because `CaptureTwistedLogs` is using `StringIO` without encoding:

``` python
    def _setUp(self):
        logs = StringIO()
        full_observer = log.FileLogObserver(logs)
        self.useFixture(_TwistedLogObservers([full_observer.emit]))
        self.addDetail(self.LOG_DETAIL_NAME,
                       Content(UTF8_TEXT, lambda: [logs.getvalue()]))
```

The last line should perhaps be:

``` python
                       Content(UTF8_TEXT, lambda: [logs.getvalue().encode("utf-8")]))
```

**Comment by rbtcollins** [comment]
I agree; care to add a test and a patch in a PR?

**Comment by allenap** [comment]
I made this change by forking from GitHub's web UI but I kind of regret it now. For one, I forgot to say "fixes XXX" in the commit message, so this issue isn't linked to the PR. Sigh. Anyway, here it is: https://github.com/testing-cabal/testtools/compare/master...allenap:patch-1

**Comment by allenap** [comment]
Ha, I hadn't actually created a PR yet. I am not strong with the GitHub today.

**Comment by allenap** [comment]
This was fixed by pull request #231.
